### PR TITLE
optimize code

### PR DESCRIPTION
**Pull request by tylitianrui**
- reuse []byte memory 
-  reduce  redundant. code

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -492,10 +492,11 @@ func (c *Client) Do(req *Request, resp *Response) error {
 			c.m = m
 		}
 	}
-	hc := m[string(host)]
+	host_str := b2s(host)
```
Comment (erikdubbelboer) [inline client.go:495]: This doesn't work. The string will be used for a map key which means the map will keep a reference to it. But that's unsafe as the underlying bytes might be modified.
See: https://go.dev/play/p/q4yRaxJuq3c

Comment (erikdubbelboer) [reply client.go:495]: What might be an improvement is only doing the `string(host)` conversion once inside `if hc == nil {`. And then doing `hc := m[b2s(host)]` even though I'm not sure if Go allocates for this string cast to access a map, you could test this.

File: args.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func (a *Args) ParseBytes(b []byte) {
 
 // String returns string representation of query args.
 func (a *Args) String() string {
-	return string(a.QueryString())
+	return b2s(a.QueryString())
```
Comment (erikdubbelboer) [inline args.go:108]: I think this is a dangerous change. There could in theory be users that currency use this string and assume it is always safe to use. This change would make this string get modified when the Args get reused. So this change would be backwards incompatible. Therefor I would prefer not to do this.

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (oschwald) [reply client.go:495]: Go has not allocated for `m[string(b)]` since 1.3 or so.

File: args.go (hunk 2 above)
Comment (tylitianrui) [reply args.go:108]: thx for your review

**Pull request closed without merging**